go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -118,6 +123,90 @@ func TestComputeFilesWithXattr(t *testing.T) {
 	}
 }
 
+func TestComputeFileDigestWithXattr(t *testing.T) {
+	xattrDgName := "user.myhash"
+	overwriteXattrDgName(t, xattrDgName)
+	tests := []struct {
+		filename     string
+		contents     string
+		xattrDgStr   string
+		expectDgStr  string
+		expectErrStr string
+	}{
+		{
+			filename:    "provide_no_xattr_will_generate_real_digest(sha256sum+real_size)",
+			contents:    "123456",
+			expectDgStr: "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92/6",
+		},
+		{
+			filename:    "provide_only_digest_hash_will_generate_digest_with_real_size",
+			contents:    "123456",
+			xattrDgStr:  "1111111111111111111111111111111111111111111111111111111111111111",
+			expectDgStr: "1111111111111111111111111111111111111111111111111111111111111111/6",
+		},
+		{
+			filename:    "valid_full_digest_(hash+size)_in_xatrr_will_be_used_directly",
+			contents:    "",
+			xattrDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/666",
+			expectDgStr: "1111111111111111111111111111111111111111111111111111111111111111/666",
+		},
+		{
+			filename:     "provide_invalid_digest_(hash_missing_digits)_will_set_md.Err",
+			contents:     "123456",
+			xattrDgStr:   "abc",
+			expectErrStr: "valid hash length is 64, got length 3 (abc)",
+		},
+		{
+			filename:     "provide_invalid_full_digest_(hash_missing_digits)_will_set_md.Err",
+			contents:     "123456",
+			xattrDgStr:   "666/666",
+			expectErrStr: "valid hash length is 64, got length 3 (666)",
+		},
+		{
+			filename:     "provide_invalid_full_digest_(extra_slash)_will_set_md.Err",
+			contents:     "123456",
+			xattrDgStr:   "///666",
+			expectErrStr: "expected digest in the form hash/size, got ///666",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.filename, func(t *testing.T) {
+			file, err := os.Create(tc.filename)
```

Comment:
better to create the file under https://pkg.go.dev/testing#B.TempDir and cleanup with t.Cleanup